File: 35-search-insert-position/search-insert-position.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        int first=0,last=nums.length-1,mid=(int)(first+last)/2;
        while(first<=last){
             mid=(first+last)/2;
            if(nums[mid]==target){
                return mid;
            }
            if(nums[mid]<target){
                first=mid+1;
            }
            else{
                last=mid-1;
            }
        }
       return first;
    }
}

File: 35-search-insert-position/search-insert-position.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var searchInsert = function(nums, target) {
    right=nums.length;
    for(let i=0;i<right;i++){
        if(nums[i]==target){
            return i
        }else if (nums[i]<target){
            continue
        }else{
            return i
        }
    }
    return right
};